.github/workflows/build-images.yaml

```diff
@@ -344,111 +348,105 @@ jobs:
-
-      - name: Login to prod ECR
-        if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
-        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
-        with:
-          registry: ${{ env.ECR_PROD }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Create multi-arch images
         env:
           ARCHS: ${{ join(fromJson(inputs.archs), ' ') }}
           TAG: ${{ inputs.tag }}
         run: |
-          images=("${IMG_CONTROLLER}" \
-          "${IMG_VXLAN_CONTROLLER}" \
-          "${IMG_RUNNER}" \
-          "${IMG_SCHEDULER}" \
-          "${IMG_AUTOSCALER_AGENT}" \
-          "${IMG_DAEMON}" \
-          "${IMG_CLUSTER_AUTOSCALER}"
+          images=(
+            "${IMG_CONTROLLER}"
+            "${IMG_VXLAN_CONTROLLER}"
+            "${IMG_RUNNER}"
+            "${IMG_SCHEDULER}"
+            "${IMG_AUTOSCALER_AGENT}"
+            "${IMG_DAEMON}"
+            "${IMG_CLUSTER_AUTOSCALER}"
           )
+
           for image in "${images[@]}"; do
-            tags=("-t" "$image:$TAG")
+            tags=("-t" "${GHCR_REGISTRY}/${image}:${TAG}")
             for arch in ${ARCHS}; do
-              tags+=("$image-$arch:$TAG")
+              tags+=("${GHCR_REGISTRY}/${image}-${arch}:${TAG}")
             done
             docker buildx imagetools create "${tags[@]}"
           done
 
-      - name: Copy all merged images to ECR
-        if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
+  generate-image-maps:
+    needs: [ build ]
+    runs-on: ubuntu-22.04
+    outputs:
+      dev: ${{ steps.generate.outputs.dev }}
+      prod: ${{ steps.generate.outputs.prod }}
+    steps:
+      - name: Harden the runner (Audit all outbound calls)
+        uses: step-security/harden-runner@4d991eb9b905ef189e4c376166672c3f2f230481 # v2.11.0
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          sparse-checkout: .github/scripts/generate_image_maps.py
+          sparse-checkout-cone-mode: false
+
+      - name: Generate Image Maps
+        id: generate
+        run: python3 .github/scripts/generate_image_maps.py
         env:
-          TAG: ${{ inputs.tag }}
-        run: |
           # note: excludes neonvm-daemon because we only need it during CI, to build compute images.
-          images=("${IMG_CONTROLLER}" \
-          "${IMG_VXLAN_CONTROLLER}" \
-          "${IMG_RUNNER}" \
-          "${IMG_SCHEDULER}" \
-          "${IMG_AUTOSCALER_AGENT}" \
-          "${IMG_CLUSTER_AUTOSCALER}")
-
-          for image in "${images[@]}"; do
-            # Remove "neondatabase/" prefix from image name
-            image_without_prefix=${image#neondatabase/}
+          COMPONENTS:
+            '["${{ env.IMG_CONTROLLER}}", "${{ env.IMG_VXLAN_CONTROLLER}}", "${{ env.IMG_RUNNER}}",
```

Comment:
nit: missed right spaces for vars:
`${{ env.IMG_CONTROLLER}}` -> `${{ env.IMG_CONTROLLER }}`